autobahn/wamp/interfaces.py

```diff
@@ -647,3 +667,70 @@ class IAuthenticator(object):
     def on_challenge(session, challenge):
         """
         """
+
+
+@public
+@six.add_metaclass(abc.ABCMeta)
+class IPayloadCodec(object):
+    """
+    WAMP payload codecs are used with WAMP payload transparency mode.
+
+    In payload transparency mode, application payloads are transmitted "raw",
+    as binary strings, without any processing at the WAMP router.
+
+    Payload transparency can be used eg for these use cases:
+
+    * end-to-end encryption of application payloads (WAMP-cryptobox)
+    * using serializers with custom user types, where the serializer and
+      the serializer implementation has native support for serializing
+      custom types (such as CBOR)
+    * transmitting MQTT payloads within WAMP, when the WAMP router is
+      providing a MQTT-WAMP bridge
+    """
```

Comment:
Could also note here that "completely custom serializers can be used if you control the whole ecosystem of clients".

Might also be worth noting the obvious: "When this is enabled, the router of course cannot provide cross-serializer transparency as it normally does."